### PR TITLE
feat(lanes): Add non-blocking iterator

### PR DESCRIPTION
**Pull request by hvanz**
For reaping and rechecking. Implements WRR scheduling.

Also moved all iterator tests to `clist_iterator_test.go`.

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: mempool/errors.go [hunk 1]
```diff
@@ -11,9 +11,6 @@ var ErrTxNotFound = errors.New("transaction not found in mempool")
 // ErrTxInCache is returned to the client if we saw tx earlier.
 var ErrTxInCache = errors.New("tx already exists in cache")
 
-// ErrLaneNotFound is returned to the client when a lane is not found.
-var ErrLaneNotFound = errors.New("lane not found in mempool")
```
Comment (jmalicevic) [inline mempool/errors.go:15]: Why are we removing this?

Comment (hvanz) [reply mempool/errors.go]: I think these are changes in the target branch.

File: mempool/clist_iterator_test.go [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+	const numLanes = 3
+	const numTxs = 100
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	var counter atomic.Int64
+	go func() {
+		waitForNumTxsInMempool(numTxs, mp)
+		fmt.Println("mempool height ", mp.height.Load())
```
Comment (jmalicevic) [inline mempool/clist_iterator_test.go:137]:
```suggestion
```

File: mempool/clist_mempool.go [hunk 3]
```diff
@@ -959,12 +1049,11 @@ func (iter *CListIterator) PickLane() types.Lane {
 			continue
 		}
 
-		if iter.counters[lane] >= uint32(lane) {
+		if iter.counters[lane] >= uint(lane) {
```
Comment (sergio-mena) [inline mempool/clist_mempool.go:1052]: Isn't this logic overlapping with what is now implemented in `NonBlockingWRRIterator) Next()` ? Could we further simplify if here?

Comment (hvanz) [reply mempool/clist_mempool.go:1052]: Probably it's overlapping. But PickLane is not in its final state. Once that code is stable we can see if we can refactor.

**Pull request merged**